mesonbuild/minstall.py

```diff
@@ -236,14 +239,31 @@ def restore_selinux_contexts() -> None:
         # If the list of files is empty, do not try to call restorecon.
         return
 
-    with subprocess.Popen(['restorecon', '-F', '-f-', '-0'],
-                          stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE) as proc:
-        out, err = proc.communicate(input=b'\0'.join(os.fsencode(f) for f in selinux_updates) + b'\0')
-        if proc.returncode != 0 and not os.environ.get('DESTDIR'):
-            print('Failed to restore SELinux context of installed files...',
-                  'Standard output:', out.decode(),
-                  'Standard error:', err.decode(), sep='\n')
+    proc, out, err = Popen_safe(['restorecon', '-F', '-f-', '-0'], (b'\0'.join(os.fsencode(f) for f in selinux_updates) + b'\0').decode())
```

Comment:
Why `.decode()` here?

Reply:
he is complaining about a Byte String otherwise

Reply:
I think it is the type annotation of `Popen_safe()` that is wrong, I think it should be `write: T.Optional[T.Union[str, T.ByteString]]`. @dcbaker is that right?

Reply:
yeah any advise is welcome :)

Reply:
`popen_safe` should always return text because it uses `universal_newlines=True` and the legacy version decodes by hand.

Reply:
@jpakkane but here it calls `decode()` on the input, not output. I think the input should remain a bytestring.

Reply:
I'd like to clarify this thread, do you want to change the arg of popen_safe to be bytestring ?

Reply:
Hmm, actually forget me, the doc says input should be string too when in text mode, which our Popen_safe does:

> If streams were opened in text mode, input must be a string. Otherwise, it must be bytes.

Reply:
But then I think `(b'\0'.join(os.fsencode(f) for f in selinux_updates) + b'\0').decode()` can be simplified to `'\0'.join(f for f in selinux_updates)`, no need to build a bytestring then decode it.